sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeLocalShuffleReader.scala

```diff
@@ -56,16 +76,25 @@ case class OptimizeLocalShuffleReader(conf: SQLConf) extends Rule[SparkPlan] {
         case e: ShuffleExchangeExec => e
       }.length
     }
-
+    // Check whether additional shuffle introduced. If introduced, revert the local reader.
```

Comment:
Now this rule converts local shuffle reader for all BroadcastHashJoinExec and then reverts all local shuffle readers if any of local shuffle reader causes additional shuffle. 

Can we just revert the local shuffle readers that cause additional shuffle and keep these not?

Reply:
This is the best, but I don't know if there is an easy way to do it.

Reply:
We can implement using revert all the local reader currently and re-optimize later when we find a better way.